scripts/tools/IndexMissingRecords.php

```diff
@@ -0,0 +1,205 @@
+        $offset = (int)$this->getOption('offset');
+        $limit = (int)$this->getOption('limit');
+        $reindex = boolval($this->getOption('reindex'));
+        $mainReport = Report::createInfo(sprintf('Resources not indexed for class %s', $class));
+        $missingIndexReport = Report::createWarning('Missing resources');
+        $mainReport->add($missingIndexReport);
+        $missingResources = [];
+        $missingResourcesIndexed = [];
+        $advancedSearch = $this->getElasticSearch();
+
+        /** @var core_kernel_classes_Resource $resource */
+        foreach ($this->search($class, $offset, $limit) as $resource) {
+            if (!$resource->exists()) {
+                continue;
+            }
+
+            if (!$this->isIndexed($advancedSearch, $this->getIndexName($class), $resource->getUri())) {
+                $missingResources[$resource->getUri()] = $resource;
+
+                $missingIndexReport->add(
+                    Report::createInfo(
+                        $resource->getUri() . ' (' . $resource->getLabel() . ')'
+                    )
+                );
+            }
+        }
+
+        if ($reindex) {
+            $reIndexedReport = Report::createSuccess('ReIndexed resources');
+            $mainReport->add($reIndexedReport);
+
+            $indexer = $this->getSyncResourceResultIndexer();
+
+            foreach ($missingResources as $resource) {
+                $reIndexedReport->add(
+                    Report::createInfo(
+                        $resource->getUri() . ' (' . $resource->getLabel() . ')'
+                    )
+                );
+
+                $indexer->addIndex($resource);
+
+                $missingResourcesIndexed[$resource->getUri()] = $resource->getLabel();
+            }
+        }
+
+        $summaryReport = Report::createSuccess('Summary');
+        $summaryReport->add(Report::createInfo('Missing resources: ' . count($missingResources)));
+        $summaryReport->add(Report::createInfo('Missing resources indexed: ' . count($missingResourcesIndexed)));
+
+        $mainReport->add($summaryReport);
+
+        return $mainReport;
+    }
+
+    private function isIndexed(ElasticSearch $search, string $index, string $uri)
+    {
+        $query = new Query($index);
+        $query->addCondition('_id:"' . $uri . '"');
+        $query->setLimit(1);
+
+        return $search->search($query)->getTotalCount() > 0;
+    }
+
+    private function getIndexName(string $classUri): ?string
+    {
+        //@TODO Remove direct call for ElasticSearch
+        return IndexerInterface::AVAILABLE_INDEXES[$classUri] ?? null;
+    }
+
+    private function search(string $classUri, int $offset, int $limit): ResultSetInterface
+    {
+        $search = $this->getComplexSearchService();
+
+        $queryBuilder = $search->query();
+
+        $criteria = $search->searchType($queryBuilder, $classUri, true);
+
+        $queryBuilder = $queryBuilder->setCriteria($criteria);
```

Comment:
Do we need to override `$queryBuilder` ?

Reply:
Not sure I got the question?

Reply:
Nevermind 😅 Already checked